aspnetcore/fundamentals/logging/index.md

```diff
@@ -869,6 +869,59 @@ The following example shows how to register filter rules in code:
 * Log level `Information` and higher.
 * All categories starting with `"Microsoft"`.
 
+## Create a custom logger
+
+To add a custom logger, add a <xref:Microsoft.Extensions.Logging.ILoggerProvider> with <xref:Microsoft.Extensions.Logging.ILoggerFactory>:
+
+```csharp
+public void Configure(
+    IApplicationBuilder app,
+    IWebHostEnvironment env,
+    ILoggerFactory loggerFactory)
+{
+    loggerFactory.AddProvider(new CustomLoggerProvider(new CustomLoggerConfiguration()));
+```
+
+The `ILoggerProvider` creates one or more `ILogger`instances. The `ILogger` instances are used by the framework to log the information.
+
+### Sample custom logger configuration
+
+The sample creates different colored console entries per log level and event ID using the following configuration type:
+
+[!code-csharp[](index/samples/3.x/CustomLogger/ColoredConsoleLogger/ColoredConsoleLoggerConfiguration.cs?name=snippet)]
+
+The preceding code sets the default level to `Warning` and the color to `Yellow`. If the `EventId` is set to 0, we will log all events.
```

Comment:
Why is the `EventId` being used to limit what is being logged?

Reply:
It's just a stupid demo, to show how to add a custom demo. Maybe some things doesn't really make sense. This logger is just coloring the log outputs. 
In that case you are able to create an different colored output for a specific event id.